### PR TITLE
feat(deque): Implement rev and rev_inplace method with comprehensive test cases

### PR DESCRIPTION
**Pull request by PingGuoMiaoMiao**
#### Changes

- Added a non-in-place `rev` method to the `deque` module, aligning with `Array.rev` behavior
- Includes extensive test coverage for edge cases and normal usage
- Maintains consistency with `Array.rev`: returns a new instance, leaves original unchanged

#### Implementation Details

- Uses `UninitializedArray` for new buffer allocation
- Correctly handles circular buffer index calculations
- Time complexity: O(n), Space complexity: O(n)

### CODE REVIEW

**Comment by PingGuoMiaoMiao** [comment]
Ask me about this PR and the above questions

File: deque/deque.mbt [hunk 1]
```diff
@@ -1546,3 +1546,76 @@ pub fn[A] get(self : T[A], index : Int) -> A? {
...
+///   inspect(dq, content="@deque.of([1, 2, 3, 4, 5])") // original deque unchanged
+/// ```
+pub fn[A] T::rev(self : T[A]) -> T[A] {
+  let len = self.len
+  let new_buf = UninitializedArray::make(len)
+  // Copy elements in reverse order
+  for i in 0..<len {
+    let src_idx = (self.head + len - i - 1) % self.buf.length()
+    new_buf[i] = self.buf[src_idx]
+  }
```
Comment (peter-jerry-ye) [inline deque/deque.mbt]: Doesn't look very effecient. This should've be implemented with two blit

Comment (PingGuoMiaoMiao) [reply deque/deque.mbt]: I don't quite understand how to use two bilts? Can you give me some more tips?

Comment (peter-jerry-ye) [reply deque/deque.mbt]: Sorry, I overlooked

File: deque/deque_test.mbt [hunk 2]
```diff
@@ -1153,3 +1153,74 @@ test "deque_binary_search_wrap_around" {
...
+  dq.rev_inplace()
+  assert_eq(dq, @deque.of([5, 4, 3, 2, 1]))
+  dq.rev_inplace()
+  assert_eq(dq, @deque.of([1, 2, 3, 4, 5]))
+
+  // Test with single element
+  let dq = @deque.of([42])
+  dq.rev_inplace()
+  assert_eq(dq, @deque.of([42]))
+}
```
Comment (peter-jerry-ye) [inline deque/deque_test.mbt:1207]: Need a test where the deque is snapped as usual.

File: deque/deque_test.mbt [hunk 3]
```diff
@@ -1153,3 +1153,74 @@ test "deque_binary_search_wrap_around" {
...
+  let even_reversed = even.rev()
+  assert_eq(even, @deque.of([1, 2, 3, 4]))
+  assert_eq(even_reversed, @deque.of([4, 3, 2, 1]))
+
+  // Test with larger deque
+  let large = @deque.of([1, 2, 3, 4, 5, 6, 7, 8, 9, 10])
+  let large_reversed = large.rev()
+  assert_eq(large, @deque.of([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]))
+  assert_eq(large_reversed, @deque.of([10, 9, 8, 7, 6, 5, 4, 3, 2, 1]))
+}
```
Comment (peter-jerry-ye) [inline deque/deque_test.mbt:1166]: Same here

File: deque/deque.mbt [hunk 4]
```diff
@@ -817,9 +817,9 @@ pub fn[A] reserve_capacity(self : T[A], capacity : Int) -> Unit {
 ///   dv.push_back(1)
 ///   dv.push_back(2)
 ///   dv.push_back(3)
-///   inspect(dv.capacity(), content="10")
+///   assert_eq(dv.capacity(), 10)
 ///   dv.shrink_to_fit()
-///   inspect(dv.capacity(), content="3")
+///   assert_eq(dv.capacity(), 3)
```
Comment (peter-jerry-ye) [inline deque/deque.mbt]: Why are these changed

**Review by peter-jerry-ye — DISMISSED** [review]

File: deque/deque.mbt (hunk 4 above)
Comment (PingGuoMiaoMiao) [reply deque/deque.mbt]: It should have been changed back.
